docs/matrix.md

```diff
@@ -213,6 +213,20 @@ tasks:
     - name: param-two
       value: $(params.bar) # array replacement from array param
 ```
+
+`Matrix.Params` supports whole array replacements from array `Parameters`.
```

Comment:
It seems like this PR allows this syntax to pass validation, but does it actually support this syntax? I don't think we want to allow this syntax without explicitly supporting it-- this could be confusing for users who attempt to use it.

Reply:
Sorry this is confusing. Prior to this PR, the functionality of whole array replacements from array `Parameters` was technically in place, but the validation logic was blocking this. By moving validation to after parameter substitution we actual unleash this functionality. 

`Matrix.Params` supports whole array replacements from array `Parameters`. This is true and support with this PR.

Matrix whole array replacements from array **results*** are not yet supported and will be addressed in a separate PR. This issue was 2/3 issues discovered by Andrew when he went to implement array replacements from results.